core/domain/exp_domain.py

```diff
@@ -135,6 +135,13 @@
 TYPE_VALID_ALGEBRAIC_EXPRESSION = 'AlgebraicExpressionInput'
 TYPE_VALID_NUMERIC_EXPRESSION = 'NumericExpressionInput'
 TYPE_VALID_MATH_EQUATION = 'MathEquationInput'
+MATH_INTERACTION_TYPES = [
+    TYPE_VALID_ALGEBRAIC_EXPRESSION,
+    TYPE_VALID_NUMERIC_EXPRESSION,
+    TYPE_VALID_MATH_EQUATION
+]
+MATH_INTERACTION_DEPRECATED_RULES = [
```

Comment:
What is the plan for fully removing these rules?

Reply:
This PR removes all the code related to these rules and also adds a state migration. We've also run an audit job to audit current lessons that might be using these rules.

Reply:
I'm fine with merging this, but we should have a long-term plan for removing these. Can you add links to the audit job? Also, until when is this constant needed?

Reply:
Added audit job link to PR description. Why do we need to remove these? There are just going to be used in the state migration functions for older explorations.